tests/istio.mk

```diff
@@ -83,35 +83,82 @@ e2e_all: | $(JUNIT_REPORT)
 	$(MAKE) e2e_simple e2e_mixer e2e_bookinfo \
 	|& tee >($(JUNIT_REPORT) > $(JUNIT_E2E_XML))
 
+# Run the e2e tests, with auth enabled. A separate target is used for non-auth.
 e2e_pilot: istioctl generate_yaml
 	go test -v -timeout 20m ./tests/e2e/tests/pilot ${TESTOPTS} -hub ${HUB} -tag ${TAG}
 
+e2e_pilot_noauth: istioctl generate_yaml
+	go test -v -timeout 20m ./tests/e2e/tests/pilot ${E2E_ARGS} -hub ${HUB} -tag ${TAG} --skip-cleanup -mixer=true -auth=disable -use-sidecar-injector=false
+
+test/minikube/auth/e2e_simple:
+	go test -v -timeout 20m ./tests/e2e/tests/simple -args --auth_enable=true \
+	  --skip_cleanup  -use_local_cluster -cluster_wide -test.v \
+	  ${E2E_ARGS} ${EXTRA_E2E_ARGS} \
+           ${TESTOPTS}
+           #|& tee >($(JUNIT_REPORT) > ${OUT_DIR}/tests/test-report-auth-simple.xml)
```

Comment:
This comment applies to all the commented code to generate junit xml.

If you don't need Junit reports just yet then remove the comment and add it back in another PR when you need them. If you do need it, then you must have `set -o pipefail;` in the same line since
* make executes each line in a subshell
* not setting pipefail will let go-junit-report mask the test failures since go-junit-report alomst always returns zero

And you need to have a order-only dependency on `$(JUNIT_REPORT)` to make sure it is installed for everyone, CI or developers. The make target `e2e_all` is a good example to follow.

Reply:
Yes, we do need junit reports - but I don't want to make it too complicated. 
I'll change the circleci config instead to run it at the end, found a cleaner way.